### PR TITLE
Deck migration/rulers layer

### PR DESCRIPTION
**Pull request by weberjavi**

### CODE REVIEW

File: libs/deck-layers/src/layers/rulers/RulersLayer.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        text: getRulerLengthLabel(ruler),
+        position: centerPoint,
+        bearing: bearing <= 0 ? 270 - bearing : 90 - bearing,
+      }
+    })
+    this.setState({ labels })
+  }
+  renderLayers() {
+    const { rulers, visible } = this.props
+    this.layers = [
```
Comment (satellitestudiodesign) [inline libs/deck-layers/src/layers/rulers/RulersLayer.ts:51]: Do we need to create a layers reference on the class? why not just returning the LayersList array directly?

File: apps/fishing-map/features/map/map-interactions.hooks.ts [hunk 2]
```diff
@@ -32,58 +33,72 @@ import {
...
+        y: info.y,
+        radius: 0,
+      })
+      if (features) {
+        setDeckLayerInteraction(features)
+      }
+      if (features?.some((feature: any) => feature.layer.id === 'RulersLayer')) {
+        const rulerPoint = features.find((feature: any) => feature.object.geometry.type === 'Point')
+        if (rulerPoint) {
+          map.setProps({ getCursor: () => 'crosshair' })
```
Comment (j8seangel) [inline apps/fishing-map/features/map/map-interactions.hooks.ts:76]: is this working for you? it doesn't for me

File: apps/fishing-map/features/map/overlays/rulers/rulers.hooks.ts [hunk 3]
```diff
@@ -68,10 +71,11 @@ const useRulers = () => {
   )
 
   const onRulerMapClick = useCallback(
-    (event: MapLayerMouseEvent) => {
+    (info: PickingInfo) => {
+      const [longitude, latitude] = info.coordinate as number[]
```
Comment (j8seangel) [inline apps/fishing-map/features/map/overlays/rulers/rulers.hooks.ts:75]: I got this on the first click, could you double check?
image

**Review by j8seangel — COMMENTED** [review]
In general it looks great! However, there is still a small delay between the end rule click and the final position of it that would be great to polish. let me know if you want to debug it together!

https://github.com/GlobalFishingWatch/frontend/assets/10500650/a4df11f6-4856-401e-b3ca-39c537f0e548

**Pull request merged**